modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/internal/search/CPDefinitionIndexer.java

```diff
@@ -745,18 +747,28 @@ protected Document doGetDocument(CPDefinition cpDefinition)
 		document.addKeyword(
 			"commerceCatalogId", commerceCatalog.getCommerceCatalogId());
 
-		int cpInstanceCount =
-			_cpInstanceLocalService.getCPDefinitionInstancesCount(
-				cpDefinition.getCPDefinitionId(),
-				WorkflowConstants.STATUS_APPROVED);
-
-		if (cpInstanceCount == 1) {
-			List<CPInstance> cpInstances = cpDefinition.getCPInstances();
+		List<CPInstance> cpInstances = cpDefinition.getCPInstances();
 
+		if (cpInstances.size() == 1) {
 			CPInstance cpInstance = cpInstances.get(0);
 
 			document.addNumber(CPField.BASE_PRICE, cpInstance.getPrice());
 		}
+		else if (!cpInstances.isEmpty()) {
+			CPInstance firstCPInstance = cpInstances.get(0);
+
+			BigDecimal lowestPrice = firstCPInstance.getPrice();
+
+			for (CPInstance cpInstance : cpInstances) {
+				BigDecimal price = cpInstance.getPrice();
```

Comment:
@alecsloan this won't work as price is not stored anymore in the product but is is stored in the base price list, I think a valid soulution is to think a contributor form the pricelist entry

Reply:
@riccardo-alberti maybe youwant to add more details in how it works now

Reply:
base prices are stored in the catalog base pricelist. so for each instance you need to get the base price from the corresponding price entry on the catalog base pricelist. in the pricelist service there is this method getCatalogBaseCommercePriceList(catalogGroupId)

Reply:
So we need to rework the price range facet to use price lists?

Reply:
i think we just need to take the price of an instance from its base price list rather than from the cpInstance itself

Reply:
Please see the newest commit. Let me know if that's not what you guys were thinking. Keep in mind this is in the indexer.

Reply:
@marco-leo, @riccardo-alberti can you please check if the changes made by Alec are correct now.

Thx